compiler/modules/types/listProj.bal

```diff
@@ -67,39 +68,44 @@ function listProjPath(Context cx, int k, Conjunction? pos, Conjunction? neg) ret
             rest = NEVER;
         }
     }
-    return listProjExclude(cx, k, members, rest, neg);
+    return listProjExclude(cx, k, members, rest, listConjunction(cx, neg));
 }
 
 // Precondition k >= 0 and members[i] not empty for all i
 // This finds the projection of e[k], excluding the list of atoms in neg
 // when the type of e is given by members and rest.
 // Based on listInhabited
 // Corresponds to phi^x in AMK tutorial generalized for list types.
-function listProjExclude(Context cx, int k, FixedLengthArray members, SemType rest, Conjunction? neg) returns SemType {
+function listProjExclude(Context cx, IntSubtype? k, FixedLengthArray members, SemType rest, ListConjunction? neg) returns SemType {
     if neg == () {
-        return listMemberAt(members, rest, k);
+        return listAtomicMemberTypeAt(members, rest, k);
     }
     else {
         int len = members.fixedLength;
-        ListAtomicType nt = cx.listAtomType(neg.atom);
+        ListAtomicType nt = neg.listType;
         int negLen = nt.members.fixedLength;
+        SemType p = NEVER;
         if len < negLen {
             if isNever(rest) {
                 return listProjExclude(cx, k, members, rest, neg.next);
             }
+            foreach int i in members.initial.length() + 1 ..< nt.members.initial.length() {
+                FixedLengthArray s = fixedArrayShallowCopy(members);
+                fixedArrayFill(s, i, rest);
+                p = union(p, listProjExclude(cx, k, members, NEVER, neg.next));
+            }
             fixedArrayFill(members, negLen, rest);
             len = negLen;
         }
         else if negLen < len && isNever(nt.rest) {
             return listProjExclude(cx, k, members, rest, neg.next);
         }
         // now we have nt.members.length() <= len
-        SemType p = NEVER;
-        foreach int i in 0 ..< len {
-            SemType ntm = listMemberAt(nt.members, nt.rest, i);
-            SemType d = diff(fixedArrayGet(members, i), ntm);
+        foreach int i in 0 ..< int:max(members.initial.length(), neg.maxInitialLen) {
+            SemType d = diff(listMemberAt(members, rest, i), listMemberAt(nt.members, nt.rest, i));
             if !isEmpty(cx, d) {
                 FixedLengthArray s = fixedArrayShallowCopy(members);
```

Comment:
I'm seeing multiple cases of fixedArrayShallowCopy then fill (or fill then set). A helper function might make things clearer.

Reply:
How about `fixedArraySetOnCopy` for the name?